packages/result/src/lib/Option/None.ts

```diff
@@ -108,7 +108,7 @@ export class OptionNone implements IOption<any> {
 	}
 
 	public filter(predicate: (value: never) => boolean): OptionNone;
```

Comment:
As well as all other places

```suggestion
	public filter(predicate: (value: never) => boolean): this;
```